modules/ob-securely-referencing-string-parameters.adoc

```diff
@@ -0,0 +1,75 @@
+// This module is included in the following assembly:
+//
+// builds/configuring-build-strategies.adoc
+
+:_content-type: REFERENCE
+[id="ob-securely-referencing-string-parameters_{context}"]
+= Securely referencing string parameters
+
+You can reference string parameters by using the `$(params.your-parameter-name)` syntax in your build strategy steps. You can use string parameters when defining environment variables, arguments, or images in a `BuildStrategy` or `ClusterBuildStrategy` CR. 
+
+[NOTE] 
+====
+You can also reference system parameters and strategy parameters by using the `$(params.your-parameter-name)` syntax in your build strategy steps. 
+====
+
+
+In the pod, all `$(params.your-parameter-name)` variables will automatically be replaced by actual strings. However, you must pay attention when you reference a string parameter in an argument by using an inline script. For example, to securely pass a parameter value into an argument defined with a script, you can choose one of the following approaches:
+
+* Using environment variables: In this approach, you pass the string parameter into an environment variable, instead of directly using it inside the script. By using quoting around the environment variable, you can avoid the command injection vulnerability. You can use this approach for strategies, such as `ko` or `buildpacks v3`. The following example uses an environment variable inside the script to reference a string parameter:
```

Comment:
```suggestion
* Using environment variables: In this approach, you pass the string parameter into an environment variable, instead of directly using it inside the script. By using quoting around the environment variable, you can avoid the command injection vulnerability. You can use this approach for strategies, such as `buildpacks v3`. The following example uses an environment variable inside the script to reference a string parameter:
```